### ISSUE TITLE
Meet requirements for deferred Clippy warnings

### ISSUE THREAD

**Issue by marcelstanley** [cartesi/rollups#202]
## 📚 Context

As first raised at #159, Clippy raises a few warnings which could not be addressed at said PR.

In the meantime, the corresponding lints listed below were disabled during CI runs:

- [*new_ret_no_self*](https://rust-lang.github.io/rust-clippy/master/index.html#/new_ret_no_self);
- [*module_inception*](https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception) - affects code generated via `state-fold-types`.

## ✔️ Solution

For, *module_inception*, we'd have the choice of suppressing the warnings via configuration or changing the business logic to meet the lint requirements. Both options would depend on #182 being completed.

In the case of *new_ret_no_self*, we would need changing the business logic to return the proper types.

## 📈 Subtasks

- [ ] Either meet requirements for *module_inception* or suppress it via linter annotations
- [ ] Meet requirements for *new_ret_no_self*

**Comment by marcelstanley** [comment]
We should also look at the [warnings that are being suppressed.](https://github.com/cartesi/rollups/pull/159#issuecomment-1649791218)
